realsense/plugin.cpp

```diff
@@ -35,17 +43,111 @@ class realsense : public plugin {
+        }
+        if (!T26X_found && !D4XXI_found){
+            std::cout << "No supported Realsense device detected!" << std::endl;
+        }
+    }
+
+    void configure_camera()
+    {
+        rs2::context ctx;
+        rs2::device_list devices = ctx.query_devices();
+        find_supported_devices(devices);
+        if (realsense_cam.compare("auto") == 0) {
+            if (D4XXI_found){
+                cam_select = D4XXI;
+                #ifndef NDEBUG
+                    std::cout << "Setting cam_select: D4XX" << std::endl;
+                #endif
+            }
+            else if (T26X_found){
+                cam_select = T26X;
+                #ifndef NDEBUG
+                    std::cout << "Setting cam_select: T26X" << std::endl;
+                #endif
+            }
+        }
+        else if ((realsense_cam.compare("D4XX") == 0) && D4XXI_found){
+            cam_select = D4XXI;
+            #ifndef NDEBUG
+                std::cout << "Setting cam_select: D4XX" << std::endl;
+            #endif
+        }
+        else if ((realsense_cam.compare("T26X") == 0) && T26X_found){
+            cam_select = T26X;
+            #ifndef NDEBUG
+                std::cout << "Setting cam_select: T26X" << std::endl;
+            #endif
+        }
+        if (cam_select == UNSUPPORTED){
+            ILLIXR::abort("Supported Realsense device NOT found!");
+        }
+        if (cam_select == T26X){
+            //T26X series has fixed options for accel rate, gyro rate, fisheye resolution, and FPS
+            cfg.enable_stream(RS2_STREAM_ACCEL, RS2_FORMAT_MOTION_XYZ32F); // 62 Hz
+            cfg.enable_stream(RS2_STREAM_GYRO, RS2_FORMAT_MOTION_XYZ32F); // 200 Hz
+            cfg.enable_stream(RS2_STREAM_FISHEYE, 1, RS2_FORMAT_Y8); //848x800, 30 FPS
+            cfg.enable_stream(RS2_STREAM_FISHEYE, 2, RS2_FORMAT_Y8); //848x800, 30 FPS
+            profiles = pipe.start(cfg, [&](const rs2::frame& frame) { this->callback(frame); });
+        }
+        else if (cam_select == D4XXI){
+            cfg.enable_stream(RS2_STREAM_ACCEL, RS2_FORMAT_MOTION_XYZ32F, ACCEL_RATE_D4XX); // adjustable to 0, 63 (default), 250 hz
+            cfg.enable_stream(RS2_STREAM_GYRO, RS2_FORMAT_MOTION_XYZ32F, GYRO_RATE_D4XX); // adjustable set to 0, 200 (default), 400 hz
+            cfg.enable_stream(RS2_STREAM_INFRARED, 1, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Y8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_INFRARED, 2, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Y8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_COLOR, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_BGR8, FPS_D4XX);
+            cfg.enable_stream(RS2_STREAM_DEPTH, IMAGE_WIDTH_D4XX, IMAGE_HEIGHT_D4XX, RS2_FORMAT_Z16, FPS_D4XX);
             profiles = pipe.start(cfg, [&](const rs2::frame& frame) { this->callback(frame); });
             profiles.get_device().first<rs2::depth_sensor>().set_option(RS2_OPTION_EMITTER_ENABLED, 0.f); // disables IR emitter
```

Comment:
I had never noticed this before. Please add a comment saying that disabling infrared allows us to use the stereo images for SLAM, but negatively affects depth quality, especially in low light conditions.